test/common/router/route_fuzz_test.cc

```diff
@@ -37,7 +51,68 @@ cleanRouteConfig(envoy::config::route::v3::RouteConfiguration route_config) {
   return clean_config;
 }
 
-// Check configuration for size and unimplemented options.
+bool validateMatcherConfig(const xds::type::matcher::v3::Matcher& matcher);
+
+bool validateOnMatchConfig(const xds::type::matcher::v3::Matcher::OnMatch& on_match) {
+  if (on_match.has_matcher() && !validateMatcherConfig(on_match.matcher())) {
+    return false;
+  }
+  if (on_match.action().typed_config().type_url().find("envoy.config.route.v3.Route") !=
```

Comment:
Just to make sure I understand this correctly, if the type isn't "envoy....Route" will the config be rejected later (either in the call to `TestUtility::validate(input)` or when creating the `ConfigImpl` object)?
If not, then the config isn't valid for this fuzz test, and I think this validation should return false.